### DISCUSSION TITLE
Refetch query externally when the query was created with enabled: false.

### DISCUSSION THREAD

**Discussion by LuisCarrilloR** [TanStack/query#804] [General] [answered]
I want to refetch a query that is in another context after some actions.

The query is declared this way:
```
const { data, refetch, isFetching } = useQuery(
    'request',
    getActivities,
    {
      cacheTime: 0,
      enabled: false,
      refetchOnMount: false,
    },
  );
```

But when I want to trigger the refetch it doesn't work.
```
await queryCache.invalidateQueries('request', {
      refetchActive: false,
      refetchInactive: true,
   });
```

Any suggestions how to handle this problem?

**Comment by wolverineks** [comment, marked as answer]
could you provide a working example on codesandbox.io?
a good place to start
https://codesandbox.io/s/react-query-starter-8i7vb?file=/src/App.tsx

**Reply by LuisCarrilloR** [reply]
Hello,  [here ](https://codesandbox.io/s/clever-lovelace-sw43f?file=/src/App.tsx)I created the example.

But it seems like the query is not re-fetching data when I invalidated the query.

**Reply by wolverineks** [reply]
thanks for the example. that was very helpful
i didnt find any documentation on it, but try
`queryCache.getQuery('queryTest').refetch()` 
instead of 
`queryCache.invalidate('queryTest')`

**Reply by wolverineks** [reply]
if the query were `enabled`, i think `queryCache.invalidate('queryTest')` would have worked

**Reply by LuisCarrilloR** [reply]
Thank you @wolverineks.

I am using typescript and it seems like refetch is not a valid method on getQuery.

queryCache.getQuery('query') returns a type CachedQuery that doesn't include the method refetch. Is this a type definition error?

**Reply by wolverineks** [reply]
im not sure, but i think it still works

**Reply by LuisCarrilloR** [reply]
I just logged the object and there is the refetch method (not included on the type definition). Thank you so much!
